### DISCUSSION TITLE
Automatically wipe the source after back up complete

### DISCUSSION THREAD

**Discussion by rawexp** [outdoorbits/little-backup-box#292] [Ideas]
Is it possible to set the lbb up to automatically wipe the source after the back up completes? My source is a camera and I would like someone else to copy the new images for me just by plugging the lbb into the camera. I would prefer as much as possible be automated so they wouldn’t have to manually format the camera’s memory card after each backup.

**Comment by outdoorbits** [comment]
This was discussed earlier. I would consider this to be risky but maybe this could be added and configurable.
It would be easy to implement this in rsync. But does gphoto2 allow to delete sources after transfer? I think it would be most secure if gphoto2 would decide to delete sources after successful transfer. Else I could implement a delete loop after backup...

**Reply by rawexp** [reply]
Thanks for getting back to me. Yes, definitely risky but (in my case) not as risky as my helper accidentally changing a camera setting when formatting the card via the camera menu. I think it could probably be done via gphoto2 although I don't know if this would only be possible on some camera models. Maybe using rsync would be easier/more widely compatible. I will look into it next time I get a chance. In the meantime please let me know if you decide to implement something. Thanks for your continued work on lbb!

**Comment by outdoorbits** [comment]
I'll need some days but I could implement this. If you know about gphoto2, is there a way to delete files automatically from source after successful transfer?

**Reply by rawexp** [reply]
I don't think there is a "download-and-then-delete" function in gphoto2 but you can delete files manually using this command:
[[-d RANGE or NAME] | [--delete-file RANGE or NAME]] [[-D] | [--delete-all-files]]

Maybe using the "delete all files" after copying has finished would be the easiest option?

**Comment by outdoorbits** [comment]
One of my concerns is that not every run is a primary backup. If anybody wants to backup his target storage, moving the files never should happen. This cold be addressed by creating a marker file in the root dir of every target when a partition is used as target the first time. This would be o.k. for me.
The other question is, can I really safely delete the source, if a file is handled by gphoto2? rsync can remove the source after successful transfer. But do I finally know, a gphoto2 transfer was successful? I can't compare the files using gphoto2 because this would be too slow. Is it safe enough just to check the size of the file at target is > 0?

**Comment by outdoorbits** [comment]
I added the option to settings/backup and to backup/modifications. My tests worked fine. Would you please test and report?

**Reply by rawexp** [reply]
Wow, quick work! Thank you. I will test it tomorrow
